govcd/api_token.go

```diff
@@ -31,6 +33,48 @@ func (vcdClient *VCDClient) SetApiToken(org, apiToken string) (*types.ApiTokenRe
 	return tokenRefresh, nil
 }
 
+func (vcdClient *VCDClient) SetServiceAccountApiToken(org, apiTokenFile string) error {
+	if vcdClient.Client.APIVCDMaxVersionIs("< 37.0") {
+		version, err := vcdClient.Client.GetVcdFullVersion()
+		if err == nil {
+			return fmt.Errorf("minimum version for Service Account authentication is 10.4 - Version detected: %s", version.Version)
+		}
+		// If we can't get the VCD version, we return API version info
+		return fmt.Errorf("minimum API version for Service Account authentication is 37.0 - Version detected: %s", vcdClient.Client.APIVersion)
+	}
+
+	apiTokenFile = filepath.Clean(apiTokenFile)
+	data, err := os.ReadFile(apiTokenFile)
+	if err != nil {
+		return err
+	}
+
+	saApiToken := &types.ApiTokenRefresh{}
+	err = json.Unmarshal(data, &saApiToken)
+	if err != nil {
+		return err
+	}
+
+	saApiToken, err = vcdClient.SetApiToken(org, saApiToken.RefreshToken)
+	if err != nil {
+		return err
+	}
+
+	data, err = json.Marshal(&types.ApiTokenRefresh{
+		RefreshToken: saApiToken.RefreshToken,
+	})
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(apiTokenFile, data, 0600)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
Should we make errors more verbose/explanatory or is it not needed?

Reply:
I added more verbose errors upon which a user can act upon. Left others as is.